tests/test_core.py

```diff
@@ -4,111 +4,158 @@
 
 from unittest.mock import patch
 
-import pytest, json, secrets, base64
+import pytest
+import json
+import secrets
+import base64
+
 
 @pytest.fixture
 def client():
     app.config['TESTING'] = True
     client = app.test_client()
 
+    # reset card mock
+    client.put('/mock', data=json.dumps({
+        'enabled': True,
+        'hasCard': False
+    }))
+
     yield client
 
     # any cleanup goes here
 
-@patch('smartcards.card.CardInterface.read', return_value=[b"XYZ", False])
-@patch('smartcards.card.CardInterface.read_long', return_value=None)
-def test_card_read(MockCardInterfaceReadLong, MockCardInterfaceRead, client):
+
+def test_card_read(client):
+    client.put('/mock', data=json.dumps({'shortValue': 'XYZ'}))
+
     rv = json.loads(client.get("/card/read").data)
     assert rv['present']
     assert rv['shortValue'] == 'XYZ'
     assert not rv['longValueExists']
-    assert MockCardInterfaceRead.called
 
     rv = json.loads(client.get("/card/read_long").data)
     assert rv == {}
-    assert MockCardInterfaceReadLong.called
 
-@patch('smartcards.card.CardInterface.read', return_value=[b'', False])
-def test_card_read_badcard(MockCardInterfaceRead, client):
+
+def test_card_read_badcard(client):
+    client.put('/mock', data=json.dumps({'shortValue': ''}))
+
     rv = json.loads(client.get("/card/read").data)
     assert rv['present']
     assert 'shortValue' not in rv
-    assert MockCardInterfaceRead.called
 
-@patch('smartcards.card.CardInterface.read', return_value=[None, False])
-def test_card_read_nocard(MockCardInterfaceRead, client):
+
+def test_card_read_nocard(client):
+    client.put('/mock', data=json.dumps({'shortValue': None}))
+
     rv = json.loads(client.get("/card/read").data)
     assert not rv['present']
     assert 'shortValue' not in rv
-    assert MockCardInterfaceRead.called
-    
+
+
 def test_card_read_no_cardreader(client):
     rv = json.loads(client.get("/card/read").data)
     assert rv['present'] == False
 
-@patch('smartcards.card.CardInterface.read', return_value=[b"XYZ", True])
-@patch('smartcards.card.CardInterface.read_long', return_value=b"Hello 1 2 3")
-def test_card_read_long(MockCardInterfaceReadLong, MockCardInterfaceRead, client):
+
+def test_card_read_long(client):
+    client.put(
```

Comment:
I can imagine abstracting out this `put` to some test helper function in case you want to eventually change how that mocking happens. But I can live with this.